### PR TITLE
[Golang] Add Integration Package with heap Data Stream

### PR DESCRIPTION
**Pull request by niraj-crest**
- Enhancement

## What does this PR do?

- Added 1 data stream (Heap metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/golang) directory.
- Run the following command to run tests. elastic-package test

## Related issues

- Relates #4864

## Screenshots
golang-heap-dashboard

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Comment by SubhrataK** [comment]
1. The line graphs and the summary for GC Count Overtime and heap Summary, are overlapping. Please move the summary outside the graph panel, as in the case of System Memory and Heap Memory Graphs.
2. Please break the GC duration over time panel into two different charts, as there is too much information displayed in just one graph

**Review by SubhrataK — APPROVED** [review]

File: packages/golang/data_stream/heap/agent/stream/httpjson.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+config_version: 2
+interval: {{period}}
+{{#if ssl}}
+request.ssl: {{ssl}}
+{{/if}}
+request.method: GET
```
Comment (ishleenk17) [inline packages/golang/data_stream/heap/agent/stream/httpjson.yml.hbs:6]: Are both the GET and POST methods available for getting metrics?

Comment (niraj-crest) [reply packages/golang/data_stream/heap/agent/stream/httpjson.yml.hbs:6]: Yes, Both are available and provide the same result.

Comment (ishleenk17) [reply packages/golang/data_stream/heap/agent/stream/httpjson.yml.hbs:6]: Any performance difference between using the 2 ?

Comment (niraj-crest) [reply packages/golang/data_stream/heap/agent/stream/httpjson.yml.hbs:6]: Nop. we get the same response from both methods.

**Comment by niraj-crest** [comment]
Due to limitation of integration package, following fields were droped 
1. golang.heap.gc.pause.count
2. golang.heap.gc.pause.avg.ns
3. golang.heap.gc.pause.sum.ns
4. golang.heap.gc.pause.max.ns

We have raised a github issue related to this limitation [here](https://github.com/elastic/integrations/issues/5171).

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**